plugins/experimental/remap_purge/remap_purge.c

```diff
@@ -0,0 +1,335 @@
+{
+  TSMBuffer reqp;
+  TSMLoc hdr_loc = NULL, url_loc = NULL;
+  bool should_purge = false;
+
+  if (TS_SUCCESS == TSHttpTxnClientReqGet(txnp, &reqp, &hdr_loc)) {
+    /* First see if we require the "secret" to be passed in a header, and then use that */
+    if (purge->header) {
+      TSMLoc field_loc = TSMimeHdrFieldFind(reqp, hdr_loc, purge->header, purge->header_len);
+
+      if (field_loc) {
+        const char *header;
+        int header_len;
+
+        header = TSMimeHdrFieldValueStringGet(reqp, hdr_loc, field_loc, -1, &header_len);
+        TSDebug(PLUGIN_NAME, "Checking for %.*s == %s ?", header_len, header, purge->secret);
+        if (header && (header_len == purge->secret_len) && !memcmp(header, purge->secret, header_len)) {
+          should_purge = true;
+        }
+        TSHandleMLocRelease(reqp, hdr_loc, field_loc);
+      }
+    } else {
+      /* We are matching on the path component instead of a header */
+      if (TS_SUCCESS == TSHttpHdrUrlGet(reqp, hdr_loc, &url_loc)) {
+        int path_len = 0, method_len = 0;
+        const char *path   = TSUrlPathGet(reqp, url_loc, &path_len);
+        const char *method = TSHttpHdrMethodGet(reqp, hdr_loc, &method_len);
+
+        TSDebug(PLUGIN_NAME, "Checking PATH = %.*s", path_len, path);
+        if (((TS_HTTP_METHOD_PURGE == method) || ((TS_HTTP_METHOD_GET == method) && purge->allow_get)) && path &&
+            (path_len >= purge->secret_len) && !memcmp(path + (path_len - purge->secret_len), purge->secret, purge->secret_len)) {
+          should_purge = true;
+        }
+        TSHandleMLocRelease(reqp, hdr_loc, url_loc);
+      }
+    }
+    TSHandleMLocRelease(reqp, TS_NULL_MLOC, hdr_loc);
+  }
+
+  /* Setup the continuation to handle this request if appropriate, if not, set the GenID if needed */
+  if (should_purge) {
+    TSCont cont = TSContCreate(purge_cont, TSMutexCreate());
+
+    TSContDataSet(cont, purge);
+    TSHttpTxnHookAdd(txnp, TS_HTTP_CACHE_LOOKUP_COMPLETE_HOOK, cont);
+    TSHttpTxnHookAdd(txnp, TS_HTTP_SEND_RESPONSE_HDR_HOOK, cont);
+  } else if (purge->gen_id > 0) {
+    TSHttpTxnConfigIntSet(txnp, TS_CONFIG_HTTP_CACHE_GENERATION, purge->gen_id);
+  }
+}
+
+TSReturnCode
+TSRemapInit(TSRemapInterface *api_info, char *errbuf, int errbuf_size)
+{
+  TSDebug(PLUGIN_NAME, "initialized");
+  return TS_SUCCESS;
+}
+
+TSReturnCode
+TSRemapNewInstance(int argc, char *argv[], void **ih, char *errbuf, int errbuf_size)
+{
+  char *id                             = argv[0]; /* The ID is default to the "from" URL, so save it */
+  PurgeInstance *purge                 = TSmalloc(sizeof(PurgeInstance));
+  static const struct option longopt[] = {
+    {(char *)"id", required_argument, NULL, 'i'},     {(char *)"secret", required_argument, NULL, 's'},
+    {(char *)"header", required_argument, NULL, 'h'}, {(char *)"state_file", required_argument, NULL, 'f'},
+    {(char *)"allow_get", no_argument, NULL, 'a'},    {NULL, no_argument, NULL, '\0'},
```

Comment:
It seems that we have a mixture of plugins that accept ``--long_option`` and ``--long-option``. Consider switching to the ``--long-option`` team 👯